go/test/endtoend/vreplication/resharding_workflows_v2_test.go

```diff
@@ -637,21 +637,58 @@ func setupCustomer2Keyspace(t *testing.T) {
 	c2shards := []string{"-80", "80-"}
 	c2keyspace := "customer2"
 	if _, err := vc.AddKeyspace(t, []*Cell{vc.Cells["zone1"]}, c2keyspace, strings.Join(c2shards, ","),
-		customerVSchema, customerSchema, defaultReplicas, defaultRdonly, 1200, nil); err != nil {
+		customerVSchema, customerSchema, 0, 0, 1200, nil); err != nil {
 		t.Fatal(err)
 	}
 	for _, c2shard := range c2shards {
 		err := cluster.WaitForHealthyShard(vc.VtctldClient, c2keyspace, c2shard)
 		require.NoError(t, err)
-		if defaultReplicas > 0 {
-			require.NoError(t, vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.replica", c2keyspace, c2shard), defaultReplicas, 30*time.Second))
-		}
-		if defaultRdonly > 0 {
-			require.NoError(t, vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.rdonly", c2keyspace, c2shard), defaultRdonly, 30*time.Second))
-		}
+		require.NoError(t, vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.primary", c2keyspace, c2shard), 1, 30*time.Second))
 	}
 }
 
+func setupMinimalCluster(t *testing.T) *VitessCluster {
+	cells := []string{"zone1"}
+
+	vc = NewVitessCluster(t, "TestBasicVreplicationWorkflow", cells, mainClusterConfig)
+	require.NotNil(t, vc)
+	defaultCellName := "zone1"
+	allCellNames = defaultCellName
+	defaultCell = vc.Cells[defaultCellName]
+
+	zone1 := vc.Cells["zone1"]
+
+	vc.AddKeyspace(t, []*Cell{zone1}, "product", "0", initialProductVSchema, initialProductSchema, 0, 0, 100, nil)
+
+	vtgate = zone1.Vtgates[0]
+	require.NotNil(t, vtgate)
+	err := cluster.WaitForHealthyShard(vc.VtctldClient, "product", "0")
+	require.NoError(t, err)
+	require.NoError(t, vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.primary", "product", "0"), 1, 30*time.Second))
+
+	vtgateConn = getConnection(t, vc.ClusterConfig.hostname, vc.ClusterConfig.vtgateMySQLPort)
+	verifyClusterHealth(t, vc)
+	insertInitialData(t)
+
+	sourceTab = vc.Cells[defaultCell.Name].Keyspaces["product"].Shards["0"].Tablets["zone1-100"].Vttablet
+
+	return vc
+}
+
+func setupMinimalCustomerKeyspace(t *testing.T) {
+	if _, err := vc.AddKeyspace(t, []*Cell{vc.Cells["zone1"]}, "customer", "-80,80-",
+		customerVSchema, customerSchema, 0, 0, 200, nil); err != nil {
+		t.Fatal(err)
+	}
+	require.NoError(t, cluster.WaitForHealthyShard(vc.VtctldClient, "customer", "-80"))
+	require.NoError(t, cluster.WaitForHealthyShard(vc.VtctldClient, "customer", "80-"))
+	require.NoError(t, vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.primary", "customer", "-80"), 1, 30*time.Second))
+	require.NoError(t, vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.primary", "customer", "80-"), 1, 30*time.Second))
+	custKs := vc.Cells[defaultCell.Name].Keyspaces["customer"]
+	targetTab1 = custKs.Shards["-80"].Tablets["zone1-200"].Vttablet
+	targetTab2 = custKs.Shards["80-"].Tablets["zone1-300"].Vttablet
+}
+
```

Comment:
These seem to rely on undeclared defaults, e.g. the keyspace names, cells, shards, etc. We could make those explicit defaults as consts.